### ISSUE TITLE
how do you document a C++ library?

### ISSUE THREAD

**Issue by mapsam** [mapbox/hpp-skel#4]
I'd love to see some default documentation suggestions for a header-only library. We could use documentation.js, but that might be odd since the JSDOC syntax is specific to JS usage, not how to use functions from a C++ lib.

cc @springmeyer @GretaCB

**Comment by mapsam** [comment]
I added a JSDOC comment to the [exclaim function](https://github.com/mapbox/hpp-skel/blob/master/include/mapbox/hello_world/exclaim.hpp#L7-L16) as a starter. I can see this working better than expected - but only for standard functional programming. Not sure how it will work with C++ classes.

``` cpp
/**
 * Add an exclamation point to the end of a string
 * @memberof hello_world
 * @name exclaim
 * @param {std::string} message
 * @returns {std::string} message!
 * @example
 * std::string value = exclaim ("hello");
 * cout << value; // => 'hello!'
 */
std::string exclaim (std::string message)
{
    std::string response = message + "!";
    return response;
}
```

**Comment by mapsam** [comment]
Going to stick with documentation.js since we already have a package.json for publishing.

We can also link to some other options if we want, such as doxygen.

**Comment by mapsam** [comment]
Unfortunately, the `{std::string}` syntax doesn't play with with docsjs. It skips over it instead of just placing an unknown type.

**Comment by springmeyer** [comment]
> Unfortunately, the {std::string} syntax doesn't play with with docsjs. It skips over it instead of just placing an unknown type.

Ugh, okay. Hrmm... starting to feel like this is the wrong approach. Probably best to next understand how http://www.stack.nl/~dimitri/doxygen/ works and probably recommend it.

**Comment by mapsam** [comment]
Sounds good @springmeyer - I'll keep digging and make updates to #6

**Comment by mapsam** [comment]
Tried [`cldoc`](https://jessevdk.github.io/cldoc/) to generate html documentation and seeing some positive results! I'm bummed there's no markdown output, and the HTML output is quite verbose, with some minified scripts and XML data sources. That being said - it's nice to be able to write incline documentation.

<img width="1061" alt="screen shot 2016-06-24 at 9 27 13 am" src="https://cloud.githubusercontent.com/assets/1943001/16343792/08406452-39ee-11e6-9449-34ef6f0e6cb9.png">

<img width="1060" alt="screen shot 2016-06-24 at 9 27 20 am" src="https://cloud.githubusercontent.com/assets/1943001/16343796/09f164ae-39ee-11e6-9421-8c0336e375fc.png">

cc @springmeyer @GretaCB

**Comment by mapsam** [comment]
Closed in #6
